packages/lib/server/repository/PrismaPhoneNumberRepository.ts

```diff
@@ -509,16 +514,65 @@ export class PrismaPhoneNumberRepository {
       }
     }
 
-    return await prisma.calAiPhoneNumber.update({
+    return await this.prismaClient.calAiPhoneNumber.update({
       where: {
         id,
       },
       data: updateData,
     });
   }
 
-  static async findByPhoneNumber({ phoneNumber }: { phoneNumber: string }) {
-    return await prisma.calAiPhoneNumber.findFirst({
+  async setInboundProviderAgentIdIfUnset({
```

Comment:
This function does too much for a repository, it belongs to a service. A repository should only be aware of one table, and shouldn't interact with multiple / throw errors / do processing; this belongs to a service.